code/game/objects/effects/effect_system/effects_smoke.dm

```diff
@@ -206,3 +206,38 @@
 
 /datum/effect_system/smoke_spread/sleeping
 	effect_type = /obj/effect/particle_effect/smoke/sleeping
+
+/////////////////////////////////////////////
+// Changeling smoke
+/////////////////////////////////////////////
+
+/obj/effect/particle_effect/smoke/changeling
+	color = "#7A61B0"
+	lifetime = 10
+
+/obj/effect/particle_effect/smoke/changeling/smoke_mob(mob/living/carbon/M)
+	if(..())
```

Comment:
make this into a guard statement instead

Reply:
still would like to see this implemented

Reply:
Made some changes.